tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/runtime/api/RuntimeAPITest.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Map;
+
+import static io.ballerina.projects.util.ProjectConstants.USER_DIR;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_HOME;
+import static org.ballerinalang.test.util.SQLDBUtils.deleteDirectory;
+
+/**
+ * Test class to test the functionality of Ballerina runtime APIs for invoking functions.
+ *
+ * @since 2201.9.0
+ */
+public class RuntimeAPITest extends BaseTest {
+
+    private static final String testFileLocation = Paths.get("src", "test", "resources", "runtime.api")
+            .toAbsolutePath().toString();
+    private static final Path javaSrcLocation = Paths.get("src", "test", "java", "org", "ballerinalang",
+            "test", "runtime", "api").toAbsolutePath();
+    private BMainInstance bMainInstance;
+
+    @BeforeClass
+    public void setup() throws BallerinaTestException {
+        bMainInstance = new BMainInstance(balServer);
+        Path sourceRoot = Paths.get(testFileLocation, "function_invocation").toAbsolutePath();
+        Map<String, String> envProperties = new HashMap<>();
+        bMainInstance.addJavaAgents(envProperties);
+        bMainInstance.runMain("build", new String[0], envProperties, new String[0], null,
+                sourceRoot.toString());
+    }
+
+    @Test
+    public void testRuntimeAPIsForBalFunctionInvocation() throws BallerinaTestException {
+        Path jarPath = Paths.get(testFileLocation, "function_invocation", "target", "bin",
+                "function_invocation.jar").toAbsolutePath();
+        List<String> compileCmdSet = new ArrayList<>();
+        compileCmdSet.add("javac");
+        compileCmdSet.add("-cp");
+        compileCmdSet.add(jarPath.toString());
+        compileCmdSet.add("-d");
+        compileCmdSet.add("targetDir");
+        compileCmdSet.add(Paths.get(javaSrcLocation.toString(), "RuntimeAPICall.java").toString());
+        ProcessBuilder compile = new ProcessBuilder(compileCmdSet).directory(javaSrcLocation.toFile());
+        try {
+            Process process = compile.start();
+            process.waitFor();
+        } catch (InterruptedException | IOException e) {
+            throw new BallerinaTestException("Error occurred while compiling the java file");
+        }
+
+        List<String> unzipProcessCmdSet = new ArrayList<>();
+        unzipProcessCmdSet.add("unzip");
+        unzipProcessCmdSet.add("-qq");
+        unzipProcessCmdSet.add(jarPath.toString());
+        unzipProcessCmdSet.add("-d");
+        unzipProcessCmdSet.add("targetDir");
+        ProcessBuilder unzipProcess = new ProcessBuilder(unzipProcessCmdSet).directory(javaSrcLocation.toFile());
+        unzipProcess.redirectErrorStream(true);
+        try {
+            Process process = unzipProcess.start();
+            process.waitFor();
+        } catch (IOException | InterruptedException e) {
+            throw new BallerinaTestException("Error occurred while unzipping the jar file");
+        }
+
+        List<String> jarCmdSet = new ArrayList<>();
```

Comment:
Shall we add some comments to explain the testing approach?

Reply:
refactored the test code